var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -173,6 +175,10 @@ class Llvm(CMakePackage, CudaPackage):
     # MLIR exists in > 10.x
     conflicts("+mlir", when="@:9")
 
+    # Code-signing variant is specific to macOS!
+    conflicts('+code_signing',    when='~macos')
+    conflicts('+code_signing',    when='~lldb')
```

Comment:
```suggestion
    conflicts('+code_signing',    when='~lldb', msg='A useful message to the user')
```

Reply:
And isn't this one missing?
```py
conflicts('~code_signing, when='+lldb platform=Darwin, msg='...')
```

Maybe we can turn `lldb` off by default for macOS? Because quite some packages depends on `llvm` without constraints, e.g. in the python sphere.

Reply:
Added conflict messages